pallets/msa/src/tests/offchain_tests.rs

```diff
@@ -126,14 +136,107 @@ pub fn offchain_worker_should_populate_accounts_with_offchain_indexed_events() {
 		Msa::offchain_worker(block_number);
 
 		// assert
-		for (i, account) in accounts.into_iter().enumerate() {
+		for (i, account) in accounts.iter().enumerate() {
 			let msa_key = get_msa_account_storage_key_name((i + 1) as MessageSourceId);
 			let result = get_index_value::<Vec<AccountId32>>(&msa_key);
-			assert_eq!(result, Ok(Some(vec![account])));
+			assert_eq!(result, Ok(Some(vec![account.clone()])));
 		}
 
 		let last_processed_block =
 			get_index_value::<BlockNumberFor<Test>>(&LAST_PROCESSED_BLOCK_STORAGE_NAME[..]);
 		assert_eq!(last_processed_block, Ok(Some(block_number)));
 	});
+
+	ext.execute_with(|| {
+		block_number += 1;
+		System::set_block_number(block_number);
+		retire_accounts::<Test>(accounts.clone());
+		let response = FinalizedBlockResponse {
+			result: "0x5685c63b9df72b59f6fa8e1223532c041d15a1abbe39a6d2a48d6565a091839d"
+				.to_string(),
+		};
+		let decoded_from_hex = hex::decode(&response.result[2..]).expect("should decode hex");
+		let val = <<Test as frame_system::Config>::Hash>::decode(&mut &decoded_from_hex[..])
+			.expect("should decode hash");
+		frame_system::BlockHash::<Test>::set(block_number, val);
+		let serialized_block = serde_json::to_string(&response).expect("should serialize");
+		let response_bytes = serialized_block.as_bytes().to_vec();
+		state.write().expect_request(PendingRequest {
+			method: "POST".into(),
+			uri: RPC_FINALIZED_BLOCK_REQUEST_URL.into(),
+			headers: vec![("Content-Type".into(), "application/json".into())],
+			sent: true,
+			body: RPC_FINALIZED_BLOCK_REQUEST_BODY.to_vec(),
+			response: Some(response_bytes),
+			..Default::default()
+		});
+	});
+
+	ext.persist_offchain_overlay();
+
+	ext.execute_with(|| {
+		// act
+		Msa::offchain_worker(block_number);
+
+		// assert
+		for (i, _) in accounts.into_iter().enumerate() {
+			let msa_key = get_msa_account_storage_key_name((i + 1) as MessageSourceId);
+			let result = get_index_value::<Vec<AccountId32>>(&msa_key);
+			assert_eq!(result, Ok(None));
+		}
+
+		let last_processed_block =
+			get_index_value::<BlockNumberFor<Test>>(&LAST_PROCESSED_BLOCK_STORAGE_NAME[..]);
+		assert_eq!(last_processed_block, Ok(Some(block_number)));
+	});
+}
+
+#[test]
+fn get_bucket_number_should_return_sudo_random_value() {
```

Comment:
I was initially very confused, but then realized you mean:
```suggestion
fn get_bucket_number_should_return_pseudo_random_value() {
```